### ISSUE TITLE
Zen mode indicator

### ISSUE THREAD

**Issue by BHydden** [online-go/online-go.com#2179]
**Is your feature request related to a problem? Please describe.**
Players want to use zen mode but don't want to appear rude by not responding in chat.

**Describe the solution you'd like**
An automated chat message sent when one player enters zen mode (perhaps only for live games?)

**Describe alternatives you've considered**
Maybe some kind of indicator light indicating zen mode is active on the player card?

**Additional context**
[Forum comment](https://forums.online-go.com/t/feature-request-zen-mode-default-option/46700/11?u=bhydden)

**Comment by csessh** [comment]
This seems simple enough, I suppose? 

Is something along this line what you're looking for? 
![Screenshot 2023-03-09 at 9 47 15 pm](https://user-images.githubusercontent.com/5171823/224008297-4b953b72-a059-412e-b522-3c7c9e6652c9.png)

**Comment by BHydden** [comment]
More or less 🤔 those two green entries into the malkovich log won't help, the opponent can't see those until after the game finishes.

"Entered" and "exited" are pretty self explanatory, what conditions do you see triggering "currently"?

**Comment by anoek** [comment]
I dunno, this seems kind of spammy and translation to other languages seems like it could be troublesome. I think some sort of "new chats" indicator on the client that is in zen mode would be more appropriate?

**Comment by BHydden** [comment]
> I think some sort of "new chats" indicator on the client that is in zen mode would be more appropriate?

Doesn't that completely violate the whole concept of zen mode?
They don't want to know that chats are there, they just want their opponent to know they aren't being rude.

**Comment by anoek** [comment]
Yeah I suppose so, so an indicator going the other way, perhaps a little note if the person starts typing that pops up _"Your opponent is in Zen mode and will be unable to see your chat messages"_ or something? That's a little easier said than done as it requires some backend support and answering the question of what do you do when there are two or more clients open, one or more of which are in zen mode?

**Comment by BHydden** [comment]
What if we keep it client side, and if a game the client is connected to while in zen mode receives a chat from the opponent, the client sends (limited to once per game by some kind of check) some auto reply such as

> I use zen mode and may not see your messages until after the game. Sorry for any inconvenience.

**Comment by anoek** [comment]
I guess that'd be ok. It doesn't address translation, and it'd be fragile, and feels like a bit hacky - but in the interim I reckon it's better than nothing.

**Comment by anoek** [comment]
That might not be the language that the viewer speaks though.

**Comment by BHydden** [comment]
Could we detect the language of the first comment and reply in that?

**Comment by benjaminpjones** [comment]
I feel like that's a lot of effort for a front-end hack 😆

**Comment by anoek** [comment]
Yeah, by the time we're talking about going to those lengths, it'd probably be lower effort to wire up some server side support for proper signaling support.
